_posts/2019-05-10-totw-98.md

```diff
@@ -0,0 +1,162 @@
+---
+title: "Tip of the Week #98: User-defined Flag Types and absl::Duration"
+layout: tips
+sidenav: side-nav-tips.html
+published: true
+permalink: tips/98
+type: markdown
+order: "098"
+---
+
+*by [Bradley White](mailto:bww@google.com)*
+
+NOTE: this tip contrasts legacy flags (of the form `DEFINE_flagname`)
+with new-style Abseil flags of the form `ABSL_FLAG()`. The legacy forms
+are now deprecated; as a result, some of this tip is historical in
+nature, but the advice on using custom flags is current.)
+
+*"Do you have a flag?" - Eddie Izzard.*
+
+## Flags of User-Defined Types
+
+The Abseil Flags library now supports flags of user-defined types.
+
+Previously a flag had to be one of the the fundamental types: `bool`,
+`int32_t`, `int64_t`, `uint64_t`, `double`, `std::string`, where a flag
+definition like ...
+
+```c++
+DEFINE_string(default_host, "www.google.com", "For requests with no host");
+```
+
+expands directly into the definition of a global variable: `FLAGS_default_host`,
+of type `std::string`, with a default value of `"www.google.com"` and the given
+description. These flags are deprecated and people should now prefer the
```

Comment:
I'm not sure we can say "deprecated" for an API in a project that we no longer maintain and has been forked ...